src/main/java/clients/VehicleTechnicalRecordsClient.java

```diff
@@ -168,6 +193,82 @@ private Response callPutVehicleTechnicalRecordsWithAlterations(String vin, Strin
+                .pathParam("systemNumber", systemNumber)
+                .log().method().log().uri().log().body()
+                .put("/vehicles/archive/{systemNumber}");
+        return response;
+    }
+
+    public Response postVehicleTechnicalRecordsForProvisional(String systemNumber, String requestBody, List<JsonPathAlteration> alterations) {
+        Response response = callPostVehicleTechnicalRecordsForProvisional(systemNumber, requestBody,alterations);
+
+        if (response.getStatusCode() == HttpStatus.SC_UNAUTHORIZED || response.getStatusCode() == HttpStatus.SC_FORBIDDEN) {
+            saveUtils();
+            response = callPostVehicleTechnicalRecordsForProvisional(systemNumber, requestBody,alterations);
+        }
+
+        return response;
+    }
+
+    private Response callPostVehicleTechnicalRecordsForProvisional(String systemNumber, String requestBody, List<JsonPathAlteration> alterations) {
+        //the only actions accepted are ADD_FIELD, ADD_VALUE, DELETE and REPLACE
+        String alteredBody = GenericData.applyJsonAlterations(requestBody, alterations);
+
+
+        Response response = given().filters(new BasePathFilter())
+                .contentType(ContentType.JSON)
+                .body(alteredBody)
+                .pathParam("systemNumber", systemNumber)
+                .log().method().log().uri().log().body()
+                .post("/vehicles/add-provisional/{systemNumber}");
+        return response;
+    }
+
+    private Response callPutVehicleTechnicalRecordsForCurrent(String systemNumber,String oldStatusCode, String requestBody, List<JsonPathAlteration> alterations) {
+        //the only actions accepted are ADD_FIELD, ADD_VALUE, DELETE and REPLACE
+        String alteredBody = GenericData.applyJsonAlterations(requestBody, alterations);
+
+
+        Response response = given().filters(new BasePathFilter())
+                .contentType(ContentType.JSON)
+                .body(alteredBody)
+                .pathParam("systemNumber", systemNumber)
+                .queryParam("oldStatusCode", oldStatusCode)
+                .log().method().log().uri().log().body()
+                .put("/vehicles/{systemNumber}");
+        return response;
+    }
+
+    public Response putVehicleTechnicalRecordsForCurrent(String systemNumber,String oldStatusCode, String requestBody, List<JsonPathAlteration> alterations) {
+        Response response = callPutVehicleTechnicalRecordsForCurrent(systemNumber,oldStatusCode, requestBody, alterations);
+
+        if (response.getStatusCode() == HttpStatus.SC_UNAUTHORIZED || response.getStatusCode() == HttpStatus.SC_FORBIDDEN) {
+            saveUtils();
+            response = callPutVehicleTechnicalRecordsForCurrent(systemNumber,oldStatusCode,requestBody, alterations);
+        }
+
+        return response;
+    }
+
+    public Response putVehicleTechnicalRecordsForArchived(String systemNumber, String requestBody, List<JsonPathAlteration> alterations) {
```

Comment:
Are different methods for each status required? Couldn't we have one method and send the status as a parameter? Or just include the status we want via a json path alteration?